bash_103_combining_commands/bash_103_combining_commands.md

```diff
@@ -0,0 +1,463 @@
+</script>
+***
+<div class = "answer">
+The output of `head -1 newfile.txt` is the first line of `newfile.txt`. Since no file named  `blue_crab.txt` previously existed, overwriting (`>`) and appending (`>>`) that output have the same result.
+</div>
+***
+
+2. What command would you type to create a new file `blue_crab.dat` that contains only the last two lines of this file?
+
+[[tail -2 newfile.txt > blue_crab.dat]]
+<script>
+  let input = "@input".trim().toLowerCase();
+  input == "tail -2 newfile.txt > blue_crab.dat" || input == "tail -2 newfile.txt >> blue_crab.dat";
+</script>
+***
+<div class = "answer">
+The output of `tail -2 newfile.txt` is the last two lines of `newfile.txt`. Since no file named  `blue_crab.dat` previously existed, overwriting (`>`) and appending (`>>`) that output have the same result.
+</div>
+***
+## Linking commands
+
+We could link commands by writing the output of one command to a file, and then running another command on that file. However if we don't actually need that intermediate file it is possible to speed up the process by passing the output of one command directly to another using a "pipe."
+
+The vertical line `|` is called a **pipe**. On American qwerty keyboards symbol `|` is located on the same key as the backslash ` \ ` and can be typed by using the `shift` key with ` \ `.
+
+<div class = "help">
+The pipe symbol is located in different places on different keyboards. If you have trouble finding it, try an internet search for "pipe symbol" along with the country or language of your keyboard.
+</div>
+
+With smaller files, the benefits of using pipes over making intermediate files are mostly from having fewer commands to type and run. But if you start working with really large datasets, maybe you are working in an "omics" field like genomics or metabolomics, creating and saving intermediate files, as well as writing and reading them, can use significant time and energy.  
+
+### Using the pipe `|`
+
+The pipe symbol `|` is like the arrow symbols `>` and `>>` in that it redirects the output of the command that precedes it. Instead of writing that output to a file, the pipe turns that output into input for the command that follows.
+
+In the previous section we learned how to sort the lines in the file `Animals.csv`. But maybe we only care about the last 3 animals listed alphabetically. With a pipe we can first `sort` and then immediately look at the `tail`. Give it a try:
+
+```
+sort Animals.csv | tail -3
+```
+
+You get the last three animals of the alphabetically sorted list: red panda, tiger, and wolf.
+
+What do you think will happen if you switch the order of `sort` and `tail`? Try it out:
+
+```
+tail -3 Animals.csv | sort
+```
+Now the blue morpho butterfly is first, instead of the red panda because the last three lines of `Animals.csv` were tiger, wolf, and blue morpho.
+
+<div class = "important">
+Something helpful about standard input and standard output here? but not too confusing?
```

Comment:
This must have gotten left off :)

Reply:
Decided to remove this. Learning about standard input and standard output would be a whole other module maybe? It feels far too "in the weeds" for this module.